error.go

```diff
@@ -41,10 +48,82 @@ type Error struct {
 	RequestID      string    `json:"request_id,omitempty"`
 	HTTPStatusCode int       `json:"status,omitempty"`
 	ChargeID       string    `json:"charge,omitempty"`
+
+	// Err contains an internal error with an additional level of granularity
+	// that can be used in some cases to get more detailed information about
+	// what went wrong. For example, Err may hold a ChargeError that indicates
+	// exactly what went wrong during a charge.
+	Err error `json:"-"`
 }
 
-// Error serializes the Error object and prints the JSON string.
+// Error serializes the error object to JSON and returns it as a string.
 func (e *Error) Error() string {
 	ret, _ := json.Marshal(e)
 	return string(ret)
 }
+
+// APIConnectionError is a failure to connect to the Stripe API.
+type APIConnectionError struct {
+	StripeErr *Error
```

Comment:
@kyleconroy Thanks for taking a look!

So what I was trying to do was just make this a reference back up to the "parent" error so that we could use its context to produce an error message from `Error()` rather than explicitly embed a struct and make its fields available here.

Now that I look at this again, I'm going to try to further reduce the interface surface area by making this field unexported as well.